network/patton/smartnode/snmp/mode/call.pm

```diff
@@ -0,0 +1,176 @@
+        { label => 'isdn-calls-connected-count', nlabel => 'isdn.calls.connected.count', set => {
+                key_values => [ { name => 'isdn_calls_connected_count' }, { name => 'display' } ],
+                output_template => 'connected calls: %d',
+                perfdatas => [
+                    { template => '%d', min => 0, unit => '', label_extra_instance => 1 }
+                ]
+            }
+        },
+        { label => 'isdn-calls-ongoing-count', nlabel => 'isdn.calls.ongoing.count', set => {
+                key_values => [ { name => 'isdn_calls_ongoing_count' }, { name => 'display' } ],
+                output_template => 'ongoing calls: %d',
+                perfdatas => [
+                    { template => '%d', min => 0, unit => '', label_extra_instance => 1 }
+                ]
+            }
+        },
+        { label => 'isdn-calls-accumulated-count', nlabel => 'isdn.calls.accumulated.count', set => {
+                key_values => [ { name => 'isdn_calls_accumulated_count' }, { name => 'display' } ],
+                output_template => 'total accumulated calls: %d',
+                perfdatas => [
+                    { template => '%d', min => 0, unit => '', label_extra_instance => 1 }
+                ]
+            }
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+    });
+
+    return $self;
+}
+
+my $gwEntry = '.1.3.6.1.4.1.1768.100.70.40.2.1';
+my $isdnPortEntry = '.1.3.6.1.4.1.1768.100.70.50.2.1';
+
+my $mappingGw = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.40.2.1.1' }, # gwDescr
+    gateway_calls_connected_count => { oid => '.1.3.6.1.4.1.1768.100.70.40.2.1.2' }, # gwCurrentConnectedCalls
+    gateway_calls_ongoing_count => { oid => '.1.3.6.1.4.1.1768.100.70.40.2.1.3' }, # gwCurrentOngoingCalls
+    gateway_calls_accumulated_count => { oid => '.1.3.6.1.4.1.1768.100.70.40.2.1.4' } # gwTotalAccumulatedCalls
+};
+my $mappingIsdn = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.50.2.1.1' }, # isdnPortDescr
+    isdn_calls_connected_count => { oid => '.1.3.6.1.4.1.1768.100.70.50.2.1.2' }, # isdnPortCurrentConnectedCalls
+    isdn_calls_ongoing_count => { oid => '.1.3.6.1.4.1.1768.100.70.50.2.1.3' }, # isdnPortCurrentOngoingCalls
+    isdn_calls_accumulated_count => { oid => '.1.3.6.1.4.1.1768.100.70.50.2.1.4' } # isdnPortTotalAccumulatedCalls
+};
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $snmp_results = $options{snmp}->get_multiple_table(oids => [
+        { oid => $gwEntry },
+        { oid => $isdnPortEntry }
+    ], nothing_quit => 1);
+
+    my $numGateway = 1;
+    my $gatewayEntry = $options{snmp}->map_instance(mapping => $mappingGw, results => $snmp_results->{$gwEntry}, instance => $numGateway);
+    while (defined($gatewayEntry->{display})) {
```

Comment:
the instance can start with another value than 1. It's not the right way to do the parsing.